src/NServiceBus.Core/Performance/MessageDurability/DetermineMessageDurabilityBehavior.cs

```diff
@@ -22,7 +21,7 @@ public override Task Invoke(OutgoingLogicalMessageContext context, Func<Task> ne
             {
                 context.AddDeliveryConstraint(new NonDurableDelivery());
 
-                context.SetHeader(Headers.NonDurableMessage,true.ToString());
+                context.Headers[Headers.NonDurableMessage] = true.ToString();
```

Comment:
Also, this would set `True` with a capital T. This isn't a problem for `bool.Parse`, but might cause serialization issues?